packages/client/src/runtime/core/extensions/wrapExtensionCallback.ts

```diff
@@ -40,7 +44,15 @@ export function wrapExtensionCallback<ResultT, ThisT, Args extends unknown[]>(
   return function (...args) {
     try {
       const result = fn.apply(this, args)
-      if (isPromiseLike(result)) {
+
+      if (isPrismaPromise(result)) {
+        const promise = createPrismaPromise((transaction) => {
+          return result.catch((error) => Promise.reject(new PrismaClientExtensionError(name, error)), transaction)
+        })
+
+        // we keep all other fields stored in the original promise
```

Comment:
What are those fields? From our `PrismaPromise` definition, only `toStringTag` is there, which would also be present on a new promise. There is also `TX_ID`, but that is added through proxy and won't be detected by `Object.assign`, i think.

Reply:
They can be literally field that isn't controlled by us. Some context: Accelerate needs to add `withAccelerateMeta` method on our PrismaPromise to execute an operation with some extra arguments. So it would be like:
```ts
await prisma.user.findMany({
        where: {id: 1454545},
        cache: true,
}).withAccelerateMeta()
```

Reply:
Got it. Maybe worth adding a test for it, if we don't have one already.